### ISSUE TITLE
Filters in summarizer.yaml not working for me

### ISSUE THREAD

**Issue by theurich** [esmf-org/esmf-test-summary#8]
I cannot figure out what I am doing wrong. I put several new filter patterns into [summarizer.yaml](https://github.com/esmf-org/esmf-test-summary/blob/71a8ed3f4dba3968c0fb2e78b2f03e2e1fb52d92/summarizer.yaml#L2). I did this some time yesterday, and there have been many rebuilds of the tables since. However, I see no effect of the filters I added. What am I missing?

**Comment by rsdunlapiv** [comment]
I noticed that yesterday and will take a look.  The summarizer.yaml syntax looks correct to me, so the filtering itself must have an issue.

**Comment by rsdunlapiv** [comment]
@theurich this should be fixed now.

One thing to note: the way this summarizer is set up now, the default behavior every time it runs is to pick up only the artifacts commits that have not yet been seen. This is an efficiency mechanism not to re-parse all of the artifacts history as the repo continues to grow in size.

The downside is that once an artifact commit has been parsed, it is now in the database and will continue to show up on the summary tables.  If summarizer.yaml is updated to exclude new hashes, we need to rebuild the database, which will re-parse all of the artifacts history, correctly excluding hashes.  There is now a checkbox "Rebuild database?" that can be selected when you manually trigger the GitHub Action.  

See here:
https://github.com/esmf-org/esmf-test-summary/actions/workflows/generate-esmf-summaries.yml

Basically, this means for the short term when summarizer.yaml is updated, we need to rebuild the database.  I'll take a look at whether this can be a less manual process, and I think we have some options for this.

**Comment by theurich** [comment]
@rsdunlapiv it's great to have this working. Also good to have the more efficient parsing mechanism in place. I do have two questions:

1. I see there are two places now that provide summarizer.yaml:
   - Under `esmf-test-summary` repo as `./summarizer.yaml`.
   - Under `esmf-test-scripts` repo as `python_scripts/test/summarizer.yaml`.
   How do these two files relate?
   Which one should a user of the test infrastructure edit?

2. I know the combo numbering is just use as a quick way of referencing. Is it the new efficient parsing/caching mechanism that has affected the previously existing alphabetical order wrt Machine? If we were to re-build the table now from scratch, would we get acron once again in first position, or is that now how it is working anymore? I know I can sort manually, and a

**Comment by rsdunlapiv** [comment]
> 1. I see there are two places now that provide summarizer.yaml:
>    - Under `esmf-test-summary` repo as `./summarizer.yaml`.
>    - Under `esmf-test-scripts` repo as `python_scripts/test/summarizer.yaml`.
>    How do these two files relate?
>    Which one should a user of the test infrastructure edit?

I am using the one under `esmf-test-scripts/python_scripts/test/summarizer.yaml` only for local testing of the test system itself.  The "real" `summarizer.yaml` is the one in the esmf-test-summary repo.  I added a comment to the first one to make that clear.

> I know the combo numbering is just use as a quick way of referencing. Is it the new efficient parsing/caching mechanism that has affected the previously existing alphabetical order wrt Machine? If we were to re-build the table now from scratch, would we get acron once again in first position, or is that now how it is working anymore? I know I can sort manually, and a

If you rebuild the database from scratch, then it will go through all of the un-filtered machines in alphabetical order, adding combos as they appear in the artifacts history.  So yes, in this case acorn (assuming it is not excluded in summarizer.yaml) would have the first set of combo numbers.

Let's say we do *not* rebuild the database, but just add a new machine named "aaa" that starts reporting.  "aaa" combos will get the next available combo number, so will not have combo numbers next to "acorn".   If we rebuild the database from scratch, then "aaa" will be processed first and get combo numbers starting with 1.  Basically, the combo numbers have no guarantee of any permanence whatsoever, especially if you rebuild the database.

Would it help to have a multi-column, default sort for the combos table?  

I still have a ticket (#6) to use permanent links for the individual combo tables, so those would not change even if you rebuilt the database from scratch.

**Comment by theurich** [comment]
Thanks for the explanation. Makes sense.
A multi-column default sort of the table would definitely help. On the other hand, once we have the permanent links, presumably being constructed similar to what we discussed before, a single column default sort on that column would be all that is needed, since i think it would practically boil down to the same sort order as the multi-column sort. So maybe it's more efficient just to do #6?

**Comment by theurich** [comment]
The default sort seems to be working now. This could probably be closed.
